analysis/src/machine_check.rs

```diff
@@ -182,8 +166,10 @@ impl TypeChecker {
             }
         }
 
-        let latch = machine.arguments.latch;
-        let operation_id = machine.arguments.operation_id;
+        let degree = machine.properties.degree;
```

Comment:
Does it make sense to use `let MachineProperties { degree, latch, ...} = machine.properties;`?

Reply:
mmm that gave me a slightly different idea: just embed the `MachineProperties` object directly

Reply:
or maybe not, will go with your suggestion

Reply:
ill keep your suggestion for this PR and try what I said in a separate PR